pkg/backup/item_backupper.go

```diff
@@ -454,30 +456,62 @@ func (ib *defaultItemBackupper) takePVSnapshot(obj runtime.Unstructured, log log
 		"ark.heptio.com/pv":     metadata.GetName(),
 	}
 
-	log.Info("Snapshotting PersistentVolume")
-	snapshotID, err := blockStore.CreateSnapshot(volumeID, pvFailureDomainZone, tags)
-	if err != nil {
-		// log+error on purpose - log goes to the per-backup log file, error goes to the backup
-		log.WithError(err).Error("error creating snapshot")
-		return errors.WithMessage(err, "error creating snapshot")
-	}
-
+	log.Info("Getting volume information")
 	volumeType, iops, err := blockStore.GetVolumeInfo(volumeID, pvFailureDomainZone)
 	if err != nil {
 		log.WithError(err).Error("error getting volume info")
 		return errors.WithMessage(err, "error getting volume info")
 	}
 
-	if ib.backupRequest.Status.VolumeBackups == nil {
-		ib.backupRequest.Status.VolumeBackups = make(map[string]*api.VolumeBackupInfo)
+	log.Info("Snapshotting PersistentVolume")
+	snapshot := newVolumeSnapshot(ib.backupRequest.Backup, volumeType, pvFailureDomainZone, location, iops)
+
+	snapshotID, createSnapErr := blockStore.CreateSnapshot(volumeID, pvFailureDomainZone, tags)
```

Comment:
Maybe it's a naming issue but what is the relation between creating a snapshot here, and the snapshot on ln 467?

Reply:
I looked further down and I see now what the code is doing, makes sense. Because the `snapshot` on ln 467 is not a "newly created" snapshot, is just an object that was setup, I'd rename that function `newVolumeSnapshot` to `volumeSnapshot` if it doesn't create a conflict.

I know `CreateSnapshot` is an existing method, but it seems confusing. It seems to create a snapshot, but there's also a create snapshot call on ln 478. What is throwing me off here as well is that nothing from the `snapshot` object from ln 467 is being used to make the call on ln 469.